code/modules/modular_computers/NTNet/_network.dm

```diff
@@ -98,8 +98,12 @@ GLOBAL_LIST_EMPTY(computer_networks)
 	return ARE_Z_CONNECTED(get_z(router.holder), get_z(D.holder))
 
 /datum/computer_network/proc/get_signal_strength(datum/extension/network_device/D)
-	if(check_connection(D))
-		return D.connection_type
+	if(!check_connection(D))
+		return 0
+	var/broadcast_strength = router.get_broadcast_strength()
```

Comment:
The only correct pattern of acquiring extensions is:
`var/datum/extension/network_device/router/router = get_extension(src, /datum/extension/network_device/router)`

Reply:
This is outside of the scope for this PR. Recommend making it an issue for a new PR.